pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -203,7 +213,29 @@ func init() {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	if vs.cfg == nil {
+		return
+	}
+
+	// Only on controller node it is required to register listeners.
+	var err error
+	vs.clientBuilder = clientBuilder
+	vs.client, err = clientBuilder.Client("vSphere-cloud-provider")
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to create kubeclient: %v", err))
```

Comment:
Shouldn't we return error from here so that we do not execute any other statements after this?

Reply:
I believe there is no way error can be passed. Initialized() method is not returning anything.

Reply:
In that case, shouldn't we just return from here?